config/src/main/java/com/typesafe/config/impl/ConfigImpl.java

```diff
@@ -335,16 +335,11 @@ public static void reloadSystemPropertiesConfig() {
     }
 
     private static AbstractConfigObject loadEnvVariables() {
-        Map<String, String> env = System.getenv();
-        Map<String, AbstractConfigValue> m = new HashMap<String, AbstractConfigValue>();
-        for (Map.Entry<String, String> entry : env.entrySet()) {
-            String key = entry.getKey();
-            m.put(key,
-                    new ConfigString.Quoted(SimpleConfigOrigin.newSimple("env var " + key), entry
-                            .getValue()));
-        }
-        return new SimpleConfigObject(SimpleConfigOrigin.newSimple("env variables"),
-                m, ResolveStatus.RESOLVED, false /* ignoresFallbacks */);
+        Properties props = new Properties();
+        props.putAll(System.getenv());
+
+        return (AbstractConfigObject) Parseable.newProperties(props,
+            ConfigParseOptions.defaults().setOriginDescription("env variables")).parse();
```

Comment:
It would be a little cleaner I think to add a static method in PropertiesParser.java akin to `fromProperties` but instead it would be `fromStringMap(Map<String,String>)`, to avoid going through the Parseable indirection and avoid creating a Properties object here.